### ISSUE TITLE
3er sesión

### ISSUE THREAD

**Issue by kbp2** [kbp2/cienciadedatos#4]
print(df.memory_usage(deep=True).sum() / (1024 ** 2), "MB en RAM")

**Comment by kbp2** [comment]
`<html><head></head><body>
Operación | Descripción | Sintaxis | Ejemplo (columna: tip)
-- | -- | -- | --
Media (mean) | Promedio de todos los valores | df["columna"].mean() | tips["tip"].mean()
Mediana (median) | Valor central del conjunto de datos | df["columna"].median() | tips["tip"].median()
Moda (mode) | Valor que más se repite | df["columna"].mode() | tips["tip"].mode()[0]
Desviación estándar | Qué tanto varían los datos respecto a la media | df["columna"].std() | tips["tip"].std()
Varianza | Medida técnica de dispersión | df["columna"].var() | tips["tip"].var()
Máximo | Valor más alto | df["columna"].max() | tips["tip"].max()
Mínimo | Valor más bajo | df["columna"].min() | tips["tip"].min()
Rango | Diferencia entre el mayor y el menor | df["columna"].max() - df["columna"].min() | tips["tip"].max() - tips["tip"].min()
Conteo (count) | Cantidad de datos (no nulos) | df["columna"].count() | tips["tip"].count()
Resumen general | Resumen estadístico | df["columna"].describe() | tips["tip"].describe()

</body></html>`

**Comment by kbp2** [comment]
"""Regresión Lineal"""

import pandas as pd
from sklearn.linear_model import LinearRegression
import matplotlib.pyplot as plt

Crear DataFrame
df = pd.DataFrame({
"Edad": [22, 25, 28, 30, 35, 40],
"Ingresos": [3000, 3200, 3500, 3700, 4000, 4200]
})

Separar variables
X = df[["Edad"]] # predictor
y = df["Ingresos"] # variable objetivo

Crear y ajustar el modelo
modelo = LinearRegression()
modelo.fit(X, y)

Mostrar coeficientes
print("Pendiente (m):", modelo.coef_[0])
print("Intercepto (b):", modelo.intercept_)

Graficar datos y predicción
plt.scatter(df["Edad"], df["Ingresos"], color="blue", label="Datos reales")
plt.plot(df["Edad"], modelo.predict(X), color="red", label="Modelo lineal")
plt.title("Regresión lineal: Edad vs Ingresos")
plt.xlabel("Edad")
plt.ylabel("Ingresos")
plt.legend()
plt.grid(True)
plt.show()

**Comment by kbp2** [comment]
"""DataSet"""


https://www.kaggle.com/datasets/spscientist/students-performance-in-exams

**Comment by kbp2** [comment]
"""KMeans"""

import pandas as pd
from sklearn.cluster import KMeans
import matplotlib.pyplot as plt
import seaborn as sns

Datos simulados
df = pd.DataFrame({
"Edad": [21, 23, 22, 24, 25, 26],
"Promedio": [8.1, 9.0, 8.5, 7.8, 9.2, 8.9]
})

Aplicar KMeans con 2 grupos
kmeans = KMeans(n_clusters=2, random_state=0)
df["Cluster"] = kmeans.fit_predict(df[["Edad", "Promedio"]])

Visualización
sns.scatterplot(x="Edad", y="Promedio", hue="Cluster", data=df, palette="Set2")
plt.title("Clustering de estudiantes")
plt.xlabel("Edad")
plt.ylabel("Promedio")
plt.show()

**Comment by kbp2** [comment]
"""Árboles de desición"""

import pandas as pd
import matplotlib.pyplot as plt
from sklearn.tree import DecisionTreeClassifier, plot_tree

df = pd.DataFrame({
"Edad": [21, 23, 22, 24, 25, 26],
"Promedio": [8.1, 6.5, 7.0, 9.2, 5.8, 7.5],
"Asesoria": [0, 1, 1, 0, 1, 0]
})
X = df[["Edad", "Promedio"]]
y = df["Asesoria"]

modelo = DecisionTreeClassifier()
modelo.fit(X, y)

plt.figure(figsize=(8,6))
plot_tree(modelo, feature_names=["Edad", "Promedio"], class_names=["No", "Sí"], filled=True)
plt.title("Árbol de decisión: ¿Asesoría?")
plt.show()

**Comment by kbp2** [comment]
"""Dataset"""

https://www.kaggle.com/datasets/spscientist/students-performance-in-exams
